Fixes AI Mech Control GBJing the AI

Jackraxxus opened the pull request:
## About The Pull Request

So when an AI entered a mech it wasnt considered an occupant it was just there, so it didnt get controls or UI or anything (cuz the add_control_flags proc only gives the controls to mobs in the occupants list) it was just there stuck unable to move or do anything really. I put this proc in so the AI can do stuff now.

I dunno if this is cleanest of fixes but it's 5am and in my testing it makes it go from not working to working so I'm claiming it is good enough (tm)

I cant really convey how it works in image form so I will go over the steps I did to implement this fix:
On master, mess about with AI dominating mechs or controlling mechs with a beacon. See how none of the AI buttons appear, WASD stops moving the camera around. The mouse changes to the correct mech style but nothing else works.

Call a bunch of procs on the AI to see what does something, add_occupant with the AI marked as the only argument works and makes everything function as intended.

Tell the code to do that.
## Why It's Good For The Game

I don't like mechs but I know some people do so it's cool for those people that it works :)
## Changelog
:cl:
fix: AI mech control beacons and malf AI dominate mech work again.
/:cl:

Jackraxxus committed:
fix

Sadboysuss commented:
Allegedly, 2 AIs can enter the same mech at the same time so you might want to test how that works with your fix.

Jackraxxus committed:
makes it work for 2

I dont like doing this it's cool to have the excuse of 'it wont let me do it' when a human tells an AI to be their gunner in their mech which sucks and honestly idk if it should be a feature but then again idk if mechs should be a feature so wcyd

Jackraxxus commented:
> Allegedly, 2 AIs can enter the same mech at the same time so you might want to test how that works with your fix.

Good catch. I've pushed changes that make it work for two. I've tested human pilot AI gunner, AI pilot human gunner, two AIs and two humans just to make sure. Still properly rejects AIs from fully occupied / solo mechs. Should probably change the rejection text but idk what to.
